enterprise/views.py

```diff
@@ -887,73 +976,19 @@ def post(self, request):
-                        else 'audit'
-                    try:
-                        enrollment_api_client.enroll_user_in_course(
-                            request.user.username,
-                            course_id,
-                            course_mode
-                        )
-                        LOGGER.info(
-                            'Created LMS enrollment for User {user} in Course {course_id} '
-                            'with License {license_uuid} in Course Mode {course_mode}.'.format(
-                                user=request.user.username,
-                                course_id=course_id,
-                                license_uuid=license_uuid,
-                                course_mode=course_mode
-                            )
-                        )
-                    except Exception as exc:    # pylint: disable=broad-except
-                        LOGGER.error(
-                            'Unable to create an LMS enrollment from the DSC view: {exc}'.format(
-                                exc=exc
-                            )
-                        )
-                        return redirect(failure_url)
-                try:
-                    self.create_enterprise_course_enrollment(request, consent_record, course_id, license_uuid)
-                except IntegrityError:
-                    error_code = 'ENTGDS009'
-                    log_message = (
-                        '[Enterprise DSC API] IntegrityError while creating EnterpriseCourseEnrollment.'
-                        'Course: {course_id}, '
-                        'Program: {program_uuid}, '
-                        'EnterpriseCustomer: {enterprise_customer_uuid}, '
-                        'User: {user_id}, '
-                        'License UUID: {license_uuid}, '
-                        'ErrorCode: {error_code}'.format(
-                            course_id=course_id,
-                            program_uuid=program_uuid,
-                            enterprise_customer_uuid=enterprise_uuid,
-                            user_id=request.user.id,
-                            license_uuid=license_uuid,
-                            error_code=error_code,
-                        )
-                    )
-                    LOGGER.exception(log_message)
+            try:
+                self._enroll_leaner_in_course(
+                    request=request,
+                    enterprise_customer=enterprise_customer,
+                    course_id=course_id,
+                    program_uuid=program_uuid,
+                    license_uuid=license_uuid)
+            except Exception:  # pylint: disable=broad-except
+                pass
             consent_record.granted = consent_provided
             consent_record.save()
 
-        return redirect(success_url if consent_provided else failure_url)
+        return redirect(success_url) if consent_provided else redirect(failure_url)
```

Comment:
nit: I don't see a reason this line should change

Reply:
because I had changed it to be something else, and then changed it back to consent_provided, and then didn't change it fully back. I'll do that.